helix-loader/src/session.rs

```diff
@@ -0,0 +1,90 @@
+use crate::{command_histfile, file_histfile, search_histfile};
+use bincode::serialize_into;
+use serde::{Deserialize, Serialize};
+use std::{
+    fs::{File, OpenOptions},
+    io::{self, BufRead, BufReader, Write},
+    path::PathBuf,
+};
+
+// TODO: should this contain a ViewPosition?
+// it would require exposing that type in a new crate, re-exporting in helix-view,
+// since this crate is a dependency of helix-view
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FileHistoryEntry {
+    path: PathBuf,
+    anchor: usize,
+    vertical_offset: usize,
+    horizontal_offset: usize,
+}
```

Comment:
@the-mikedavis Is there currently a place to put types shared between crates? If not, is it ok if I make a helix-types crate for this purpose?

Reply:
`_-types` crates can allow types to leak all over a project when they should be local to a specific crate. Here I think it's a good indicator that `helix-loader` is currently trying to do too much. The types should be moved into `helix-view` in a new `session` module. `helix-loader` can expose functions that give the file history file (and any other files) and then `helix-view` can read/write from those. `helix-loader` is already doing things 'above its paygrade' here by deciding what to do based on register characters: that decision should be made by a higher up crate like `helix-view`

Reply:
> `helix-loader` is already doing things 'above its paygrade' here by deciding what to do based on register characters: that decision should be made by a higher up crate like `helix-view`

I'll make this change.

About the types, my thought was that abstractions are going to leak anyway, either `-loader` needs to know about `-view` and `-term` types, or `-view` and `-term` need to know about serialisation formats. IMO it's better to leak into the smaller crate, and better to leak types than functionality (especially if that type is already public, as is the case for ViewPosition, which seems to be used extensively in `-term`), but if that's not norm for helix I'll adapt. 

What's the more relevant guideline when something conceptually spans two crates in your opinion:
- it should go in the crate higher in the dependency graph
- it should go in the crate with the revelevant types, splitting if necessary

Reply:
I think it's fine for `helix-view` and `helix-term` to be aware of serialization. `helix-loader` could re-export or wrap the necessary parts of bincode if we want some separation. The closest comparable code we have is probably config (see `Config` in `helix-view/src/editor.rs`). For config we also need to read from files and (de)serialize types that have some editor specifics like the statusline config. For that, `helix-loader` is only providing the files while the higher level crates take care of (de)serialization.

Generally I would prefer to see things in the higher crate in the dependency graph. I don't want to see editor specifics like `ViewPosition` in low-level crates like `helix-loader`.